public/app/features/geo/format/utils.ts

```diff
@@ -48,6 +50,47 @@ export function pointFieldFromLonLat(lon: Field, lat: Field): Field<Point> {
   };
 }
 
+//TODO return type Field<Geometry | undefined>;
+export function imageFieldFromLonLat(lon: Field, lat: Field, src: Field): Field<Feature> {
+  const buffer = new Array<Feature>(lon.values.length);
+  for (let i = 0; i < lon.values.length; i++) {
+    const longitude = lon.values.get(i);
+    const latitude = lat.values.get(i);
+    const imageSource = src.values.get(i);
+
+    // TODO: Add unit tests to thoroughly test out edge cases
+    // If longitude or latitude are null, don't add them to buffer
+    if (longitude === null || latitude === null) {
+      continue;
+    }
+
+    const imageStyle = new Style({
+      image: new Photo({
```

Comment:
The `Photo` thing is too specific to live in this general utils, lets put this all in the one photoLayer file